tests/dogtag/pytest-ansible/pytest/performance_test/test_cert_revocation.py

```diff
@@ -0,0 +1,68 @@
+"""
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Description: Load Test client Server Side configuration
+#   Steps :-
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Author: Deepak Punia <dpunia@redhat.com>
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Copyright Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+"""
+import threading
+import argparse
+from pki.client import PKIConnection
+from pki.cert import CertClient
+import pickle
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--hostname", help="<CA hostname>")
+parser.add_argument("--port", help="<CA Port Number>")
+parser.add_argument("--client-cert", help="path for admin.pem certificate")
+parser.add_argument("--cert_sn_file", help="path for certificate serial number file")
+
+args = parser.parse_args()
+
+# Create a PKIConnection object that stores the details of the CA.
+connection = PKIConnection('https', args.hostname, args.port)
+
+# The pem file used for authentication. Created from a p12 file using the
+# command -
+# openssl pkcs12 -in <p12_file_path> -out /tmp/auth.pem -nodes
+connection.set_authentication_cert(args.client_cert)
+
+# Instantiate the CertClient
+cert_client = CertClient(connection)
+
+
+def run_test(cert_sn):
+    # execute the specified number of tests
+    try:
+        revoke_data = cert_client.revoke_cert(cert_sn, revocation_reason='Key_Compromise')
+        if revoke_data.operation_result != 'success':
+            raise Exception("Cert enrollment failed : {}".format(revoke_data.request_id))
+    except Exception as IndexError:
+        print(IndexError)
+
+
+if __name__ == "__main__":
+    # get test parameters from CLI parameters
+    # execute the test_cert_enrollement.py script and store Serial Number data into file
+    # provide that file path to this script by using --cert_sn_file <file path>
+    cert_sn_file = open(args.cert_sn_file, 'r')
+
+    # create the specified number of threads
+    threads = []
+
+    for cert_sn in pickle.load(cert_sn_file):
+        t1 = threading.Thread(target=run_test, args=(cert_sn,))
+        t1.start()
+        threads.append(t1)
```

Comment:
IIUC this code is creating a thread for each serial number in the file, which could be a lot (e.g. 2M threads).
I think we want to do it like in the enrollment test, so we should create a set of threads (specified with `--number-of-clients`), then each thread should revoke a set of serial numbers (specified with `--number-of-tests-per-client`). So the serial numbers should be loaded into a list first, then we split the list for each thread.

Reply:
As suggested updated sub_list logic for revocation.